modules/integration/tests-integration/lazy-loading-test/src/test/java/org/wso2/appserver/integration/lazy/loading/artifacts/CarbonAppGhostDeploymentTestCase.java

```diff
@@ -0,0 +1,476 @@
+                "is not loaded. Tenant:" + TENANT_DOMAIN_2);
+        assertEquals(webAppStatusTenant2WebApp1.isWebAppStarted(), true, "Web-App: " + CARBON_APP1_WEB_APP_FILE +
+                " is not started after deployment in Tenant:" + TENANT_DOMAIN_2);
+        assertEquals(webAppStatusTenant2WebApp1.isWebAppGhost(), false, "Web-App: " + CARBON_APP1_WEB_APP_FILE +
+                " is in ghost mode after deployment in Tenant:" + TENANT_DOMAIN_2);
+
+
+        assertTrue(isCarbonAppListed(CARBON_APP_NAME1), "Carbon Application is not listed :" + CARBON_APP_NAME1);
+
+        carbonAppClient.uploadCarbonAppArtifact(CARBON_APP_FILE2, carbonApp2URLDataHandler);
+        assertTrue(WebAppDeploymentUtil.isWebApplicationDeployed(backendURL, sessionCookie, CARBON_APP2_WEB_APP_NAME),
+                "Web Application deployment failed: " + CARBON_APP2_WEB_APP_NAME + "on " + TENANT_DOMAIN_2);
+
+        WebAppStatus webAppStatusTenant2WebApp2 = getWebAppStatus(TENANT_DOMAIN_2, CARBON_APP2_WEB_APP_FILE);
+        assertEquals(webAppStatusTenant2WebApp2.getTenantStatus().isTenantContextLoaded(), true, " Tenant Context" +
+                " is not loaded. Tenant:" + TENANT_DOMAIN_2);
+        assertEquals(webAppStatusTenant2WebApp2.isWebAppStarted(), true, "Web-App: " + CARBON_APP2_WEB_APP_FILE +
+                " is not started after deployment in Tenant:" + TENANT_DOMAIN_2);
+        assertEquals(webAppStatusTenant2WebApp2.isWebAppGhost(), false, "Web-App: " + CARBON_APP2_WEB_APP_FILE +
+                " is in ghost mode after deployment in Tenant:" + TENANT_DOMAIN_2);
+
+
+        assertTrue(isCarbonAppListed(CARBON_APP_NAME2), "Carbon Application is not listed :" + CARBON_APP_NAME2);
+        log.info("Carbon application deployment end");
+    }
+
+    @Test(groups = "wso2.as.lazy.loading", description = "  Invoke web application that is deployed as Carbon " +
+            "application in Ghost Deployment enable environment.First test will restart the server gracefully.After " +
+            "the restart  all   tenant context not be loaded.Then,  it invokes the first web app on first tenant." +
+            " After the invoke, only that web app should loaded fully.",
+            dependsOnMethods = "carApplicationUploadInGhostDeployment")
+    public void testInvokeWebAppInCarbonAppInGhostDeployment() throws Exception {
+
+        serverManager.restartGracefully();
+
+
+        assertEquals(getTenantStatus(TENANT_DOMAIN_1).isTenantContextLoaded(), false, " Tenant Name:" +
+                TENANT_DOMAIN_1 + "loaded before access.");
+        assertEquals(getTenantStatus(TENANT_DOMAIN_2).isTenantContextLoaded(), false, " Tenant Name:" +
+                TENANT_DOMAIN_2 + "loaded before access.");
+
+        HttpResponse httpResponse = HttpURLConnectionClient.sendGetRequest(tenant1WebApp1URL, null);
+        assertEquals(httpResponse.getData(), WEB_APP1_RESPONSE,
```

Comment:
@warunap - IMO having comments for each assert is not required. There should be comments to explain the login in the code.. but assets should be self explanatory.